### ISSUE TITLE
Support Partial Indexes

### ISSUE THREAD

**Issue by josh-feldman** [oguimbal/pg-mem#89]
Sorry to add another one to your plate 😞 . When trying to run a partial unique index:
```sql
CREATE UNIQUE INDEX my_idx ON my_table (col1, (col2 IS NULL)) WHERE col2 IS NULL;
```

We get the following error:

```
failed to run migrations QueryError: 💔 Your query failed to parse.
    This is most likely due to a SQL syntax error. However, you might also have hit a bug, or an unimplemented feature of pg-mem.
    If this is the case, please file an issue at https://github.com/oguimbal/pg-mem along with a query that reproduces this syntax error.
    
    👉 Failed query:
    
        
            CREATE UNIQUE INDEX my_idx ON my_table (col1, (col2 IS NULL)) WHERE col2 IS NULL;
        ;;
    
    💀 Syntax error at line 2 col 71:
    
              CREATE UNIQUE INDEX my_idx ON my_table (col1, (col2 IS NULL)) WHERE
                                                                            ^
    Unexpected kw_where token: "where". Instead, I was expecting to see one of the following:
    
        - A "semicolon" token
    
    
        at Object.parseSql (/.../node_modules/pg-mem/src/parse-cache.ts:53:15)
        at DbSchema.parse (/.../node_modules/pg-mem/src/schema.ts:80:16)
        at DbSchema.queries (/.../node_modules/pg-mem/src/schema.ts:88:31)
        at queries.next (<anonymous>)
        at DbSchema.query (/.../node_modules/pg-mem/src/schema.ts:67:20)
        at MemPg.query (/.../node_modules/pg-mem/src/adapters.ts:112:76)
      data: {
        error: '💔 Your query failed to parse.\n' +
          'This is most likely due to a SQL syntax error. However, you might also have hit a bug, or an unimplemented feature of pg-mem.\n' +
          'If this is the case, please file an issue at https://github.com/oguimbal/pg-mem along with a query that reproduces this syntax error.\n' +
          '\n' +
          '👉 Failed query:\n' +
          '\n' +
          '    \n' +
          '        CREATE UNIQUE INDEX my_idx ON my_table (col1, (col2 IS NULL)) WHERE col2 IS NULL;\n' +
          '    ;;\n' +
          '\n' +
          '💀 Syntax error at line 2 col 71:\n' +
          '\n' +
          '          CREATE UNIQUE INDEX my_idx ON my_table (col1, (col2 IS NULL)) WHERE\n' +
          '                                                                        ^\n' +
          'Unexpected kw_where token: "where". Instead, I was expecting to see one of the following:\n' +
          '\n' +
          '    - A "semicolon" token\n' +
          '\n'
      }
    }
```

**Comment by oguimbal** [comment]
No worries :)

I'll try to have a look asap

**Comment by abadyan-vonage** [comment]
Bumped into this myself, so upvote :)

**Comment by oguimbal** [comment]
Should be fixed with `pg-mem@2.0.1` !

**Comment by abadyan-vonage** [comment]
Incredible customer support! 👯 
Works for me.
